app/storage_strategies/aws_s3.py

```diff
@@ -0,0 +1,64 @@
+import boto3
+import os
+from string import Template
+from botocore.exceptions import NoCredentialsError, ClientError
+from storage_strategies.storage_strategy import StorageStrategy
+
+class AWSS3StorageStrategy(StorageStrategy):
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.bucket_name = self._resolve_placeholder(context['settings'].get('bucket_name'))
+        self.region = self._resolve_placeholder(context['settings'].get('region'))
+        self.access_key = self._resolve_placeholder(context['settings'].get('access_key'))
+        self.secret_access_key = self._resolve_placeholder(context['settings'].get('secret_access_key'))
+
+        self.s3_client = boto3.client(
+            's3',
+            aws_access_key_id=self.access_key,
+            aws_secret_access_key=self.secret_access_key,
+            region_name=self.region
+        )
+
+    def _resolve_placeholder(self, value):
+        if not value:
+            return None
+        return Template(value).substitute(os.environ)
```

Comment:
Nice one! We'll need to apply it in the other storage strategies as well, great idea 🙏

Reply:
Maybe we should move it to StorageStrategy?

Reply:
Okay, I moved it and even improved it, so now it properly handles missing keys.

`Exception: Environment variable 'AWS_S3_BUCKET_NAME' is missing. Please set it in your environment or configuration.
`